Sort LL using merge sort

```diff
@@ -0,0 +1,128 @@
+#include <bits/stdc++.h> 
+using namespace std; 
+
+// Link list node 
+class node { 
```

Comment:
Class/Struct names should be in UpperCamelCase.
Follow https://google.github.io/styleguide/cppguide.html#General_Naming_Rules